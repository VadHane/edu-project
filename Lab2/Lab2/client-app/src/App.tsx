import React, {FunctionComponent} from 'react';
import {BrowserRouter} from 'react-router-dom';
import UserSTable from './components/UsersTable/UsersTable';
import './App.css';

const App: FunctionComponent = () => {
  return (
    <BrowserRouter>
      <div className="">
        <UserSTable />

      </div>
    </BrowserRouter>
  );
};

export default App;
