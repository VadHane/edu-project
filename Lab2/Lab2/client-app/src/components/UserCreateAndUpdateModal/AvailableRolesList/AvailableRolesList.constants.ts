export const GET_CREATE_ROLE_MESSAGE = 'Would you like to create new role?';

export const ADD_ROLE_MESSAGE = 'Would you like to add this role?';

export const LIST_INCLUDES_ROLE_MESSAGE = 'Roles list includes this role.';
