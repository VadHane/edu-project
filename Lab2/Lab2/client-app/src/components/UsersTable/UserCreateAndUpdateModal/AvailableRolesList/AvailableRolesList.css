img {
    height: 10px;
    margin-left: 10px;
}

span {
    font-size: small;
}