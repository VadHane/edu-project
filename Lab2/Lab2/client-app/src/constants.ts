export const USER_PICTURE_URL = 'https://cdn-icons-png.flaticon.com/512/1077/1077114.png';
export const REMOVE_IMAGE_URL = 'https://cdn-icons-png.flaticon.com/512/3096/3096687.png';
export const EDIT_IMAGE_URL = 'https://cdn-icons-png.flaticon.com/512/61/61456.png';
export const APPROVE_IMAGE_URL =
    'https://cdn-icons-png.flaticon.com/512/1828/1828743.png';
export const CANSEL_IMAGE_URL = 'https://cdn-icons-png.flaticon.com/512/1828/1828939.png';

export const LENGTH_OF_NAME_EXCEPTION =
    'Length of first name can not be less than 3 symbols.';
export const LENGTH_OF_SURNAME_EXCEPTION =
    'Length of last name can not be less than 3 symbols.';
export const INCORRECT_EMAIL_EXCEPTION = 'Incorrect email.';
export const FILE_NOT_IMAGE_EXCEPTION = 'The file must be image.';
