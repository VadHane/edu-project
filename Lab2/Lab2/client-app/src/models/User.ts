import {Role} from './Role';

export interface User {
    id: string,
    firstName: string,
    lastName: string,
    email: string,
    imageBlobKey: string,
    roles: Array<Role>,
};
