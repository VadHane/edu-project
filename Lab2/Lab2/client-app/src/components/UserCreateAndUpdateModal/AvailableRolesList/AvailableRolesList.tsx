/* eslint-disable indent */
import React, { FunctionComponent, useEffect, useState } from 'react';
import { Role } from '../../../models/Role';
import AvailableRolesListProps from './AvailableRolesList.types';
import { APPROVE_IMAGE, CANSEL_IMAGE } from '../../../constants';
import './AvailableRolesList.css';

const AvailableRolesList: FunctionComponent<AvailableRolesListProps> = (
    props: AvailableRolesListProps,
) => {
    const [roles, setRoles] = useState<Array<Role>>(props.roles);
    const [inputNameOfRole, setInputNameOfRole] = useState<string>('');
    const [isRoleNew, setItNewRole] = useState<boolean>(false);
    const [isRoleAdded, setItAddedRole] = useState<boolean>(false);

    useEffect(() => {
        setRoles([...props.roles]);
    }, [props.roles]);

    useEffect(() => {
        const availableInList: boolean =
            !roles.find((role: Role) => role.name === inputNameOfRole) &&
            inputNameOfRole.length > 1;

        const added: boolean = props.addedRoles.find(
            (role: Role) => role.name === inputNameOfRole,
        )
            ? true
            : false;

        setItNewRole(availableInList);
        setItAddedRole(added);
    }, [inputNameOfRole, props.addedRoles, roles]);

    const inputBoxNode: React.ReactNode = (
        <div>
            <datalist id="available-roles-list">
                {roles?.map((role: Role) => (
                    <option value={role.name} key={role.id}></option>
                ))}
            </datalist>
            <input
                type="text"
                list="available-roles-list"
                value={inputNameOfRole}
                maxLength={30}
                onChange={(e) => setInputNameOfRole(e.target.value)}
            />
        </div>
    );

    const approveAddingNewRoleNode: React.ReactNode = (
        <div className="aprove-add-role">
            <span>Would you like to create new role - {inputNameOfRole}?</span> <br />
            <img
                src={APPROVE_IMAGE.URL}
                alt={APPROVE_IMAGE.ALT}
                onClick={() => {
                    props.createNewRole({ id: '', name: inputNameOfRole });
                    setItNewRole(false);
                }}
            />
            <img
                src={CANSEL_IMAGE.URL}
                alt={CANSEL_IMAGE.ALT}
                onClick={() => setInputNameOfRole('')}
            />
        </div>
    );

    const approveAddingAvailableRoleNode: React.ReactNode = (
        <div className="aprove-add-role">
            <span>Would you like to add this role?</span> <br />
            <img
                src={APPROVE_IMAGE.URL}
                alt={APPROVE_IMAGE.ALT}
                onClick={() => {
                    props.addRole(getRoleByName(inputNameOfRole));
                    setInputNameOfRole('');
                }}
            />
            <img
                src={CANSEL_IMAGE.URL}
                alt={CANSEL_IMAGE.ALT}
                onClick={() => setInputNameOfRole('')}
            />
        </div>
    );

    const approveAddingNode: React.ReactNode = (
        <>{!isRoleNew ? approveAddingAvailableRoleNode : approveAddingNewRoleNode}</>
    );

    const roleWasAddedNode: React.ReactNode = (
        <div>
            <span>Roles list includes this role.</span> <br />
            <img
                src={APPROVE_IMAGE.URL}
                alt={APPROVE_IMAGE.ALT}
                onClick={() => setInputNameOfRole('')}
            />
        </div>
    );

    const getRoleByName = (roleName: String): Role => {
        const index = roles.findIndex((role: Role) => role.name === roleName);
        return roles[index];
    };

    return (
        <div>
            {inputBoxNode}
            {isRoleAdded
                ? roleWasAddedNode
                : inputNameOfRole.length > 1
                ? approveAddingNode
                : ''}
        </div>
    );
};

export default AvailableRolesList;
