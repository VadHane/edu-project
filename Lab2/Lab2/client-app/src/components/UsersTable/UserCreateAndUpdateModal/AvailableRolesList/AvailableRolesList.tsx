import React, {FunctionComponent, useEffect, useState} from 'react';
import {Role} from '../../../../models/Role';
import AvailableRolesListProps from './AvailableRolesList.types';
import './AvailableRolesList.css';

const AvailableRolesList: FunctionComponent<AvailableRolesListProps> =
(props: AvailableRolesListProps) => {
  const [roles, setRoles] = useState<Array<Role>>(props.roles);
  const [inputNameOfRole, setInputNameOfRole] = useState<string>('');
  const [itNewRole, setItNewRole] = useState<boolean>(false);
  const [itAddedRole, setItAddedRole] = useState<boolean>(false);

  useEffect(() => {
    setRoles([...props.roles]);
  }, [props.roles.length]);

  useEffect(() => {
    let availableInList: boolean = !roles.find(
        (role: Role) => role.name === inputNameOfRole);
    availableInList = availableInList && inputNameOfRole.length > 1;

    const added: boolean = props.addedRoles.find(
        (role: Role) => role.name === inputNameOfRole) ?
        true : false;

    setItNewRole(availableInList);
    setItAddedRole(added);
  }, [inputNameOfRole]);

  const inputBox: React.ReactNode = (
    <div>
      <datalist id="available-roles-list">
        {roles?.map((role: Role) => (
          <option value={role.name} key={role.id}></option>
        ))}
      </datalist>
      <input type="text" list='available-roles-list'
        value={inputNameOfRole}
        maxLength={30}
        onChange={(e) => setInputNameOfRole(e.target.value)}/>
    </div>
  );

  const aproveAddingNewRole: React.ReactNode = (
    <div className='aprove-add-role'>
      <span>Would you like to create new role - {inputNameOfRole}?</span> <br />
      <img src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
        alt="Aprove"
        onClick={() => {
          debugger;
          props.createNewRole({id: '', name: inputNameOfRole});
          setItNewRole(false);
        }}/>
      <img
        src="https://cdn-icons-png.flaticon.com/512/1828/1828939.png"
        alt="Cancel"
        onClick={() => setInputNameOfRole('')}/>
    </div>
  );

  const approveAddingAvailableRole: React.ReactNode = (
    <div className='aprove-add-role'>
      <span>Would you like to add this role?</span> <br />
      <img
        src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
        alt="Aprove"
        onClick={() => {
          props.addRole(getRoleByName(inputNameOfRole));
          setInputNameOfRole('');
        }}/>
      <img
        src="https://cdn-icons-png.flaticon.com/512/1828/1828939.png"
        alt="Cancel"
        onClick={() => setInputNameOfRole('')}/>
    </div>
  );

  const roleWasAdd: React.ReactNode = (
    <div>
      <span>Roles list consist this role.</span> <br />
      <img
        src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
        alt="Ok"
        onClick={() => setInputNameOfRole('')}/>
    </div>
  );

  const getRoleByName = (roleName: String): Role => {
    const index = roles.findIndex((role: Role) => role.name === roleName);
    return roles[index];
  };

  return (
    <div>
      {inputBox}
      {itAddedRole ? roleWasAdd :
        inputNameOfRole.length > 1 ?
          !itNewRole ? approveAddingAvailableRole :
          aproveAddingNewRole : ''}
    </div>
  );
};

export default AvailableRolesList;
