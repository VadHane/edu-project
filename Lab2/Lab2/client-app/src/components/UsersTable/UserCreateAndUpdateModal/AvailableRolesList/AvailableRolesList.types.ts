import { Role } from '../../../../models/Role';

type AvailableRolesListProps = {
    roles: Array<Role>;
    addedRoles: Array<Role>;
    addRole: (role: Role) => void;
    createNewRole: (role: Role) => void;
};

export default AvailableRolesListProps;
