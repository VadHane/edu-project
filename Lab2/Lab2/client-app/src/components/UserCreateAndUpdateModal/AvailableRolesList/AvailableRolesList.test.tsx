import React from 'react';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import { Role } from '../../../models/Role';
import AvailableRolesList from './AvailableRolesList';
import { APPROVE_IMAGE, CANCEL_IMAGE } from '../../../App.constants';
import {
    ADD_ROLE_MESSAGE,
    GET_CREATE_ROLE_MESSAGE,
    LIST_INCLUDES_ROLE_MESSAGE,
} from './AvailableRolesList.constants';

const mockedAddRole = jest.fn();
const mockedCreateRole = jest.fn();
const testRolesList: Array<Role> = [
    {
        id: '1234',
        name: 'TestRole1',
    },
    {
        id: '5678',
        name: 'TestRole2',
    },
];
const testAddedRolesList: Array<Role> = [
    {
        id: '0000',
        name: 'TestAddedRole3',
    },
];

describe('Test available role list', () => {
    afterEach(cleanup);

    test('Component should render without errors.', () => {
        render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        expect(screen.getByRole('combobox')).toBeInTheDocument();
    });

    test('Component should render correct without data.', () => {
        render(
            <AvailableRolesList
                roles={[]}
                addedRoles={[]}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        expect(screen.getByRole('combobox')).toBeInTheDocument();
    });

    test('Component should handle typing action.', () => {
        render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        const testInputValue = 'test_value_1234';

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testInputValue },
        });

        expect(screen.getByDisplayValue(testInputValue)).toBeInTheDocument();
    });

    test('Component should include message about creating new role, if there is inputted name of role, witch dont contains in rolles list.', () => {
        render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        const testInputValue = 'test_value_1234';

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testInputValue },
        });

        expect(screen.getByText(GET_CREATE_ROLE_MESSAGE)).toBeInTheDocument();
        expect(screen.getByRole('img', { name: APPROVE_IMAGE.ALT })).toBeInTheDocument();
        expect(screen.getByRole('img', { name: CANCEL_IMAGE.ALT })).toBeInTheDocument();
    });

    test('Component should not include message about creating new role, if inputted length of name of role is less then 2 symbols.', () => {
        render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        const testInputValue = 't';

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testInputValue },
        });

        expect(screen.queryByText(GET_CREATE_ROLE_MESSAGE)).toBeNull();
        expect(screen.queryByRole('img', { name: APPROVE_IMAGE.ALT })).toBeNull();
        expect(screen.queryByRole('img', { name: CANCEL_IMAGE.ALT })).toBeNull();
    });

    test('Component should include message about adding new role, if there is inputted name of role, witch contains in rolles list.', () => {
        render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testRolesList[0].name },
        });

        expect(screen.getByText(ADD_ROLE_MESSAGE)).toBeInTheDocument();
        expect(screen.getByRole('img', { name: APPROVE_IMAGE.ALT })).toBeInTheDocument();
        expect(screen.getByRole('img', { name: CANCEL_IMAGE.ALT })).toBeInTheDocument();
    });

    test('Component should include warning message about adding new role, witch was added in past.', () => {
        render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testAddedRolesList[0].name },
        });

        expect(screen.getByText(LIST_INCLUDES_ROLE_MESSAGE)).toBeInTheDocument();
        expect(screen.getByRole('img', { name: APPROVE_IMAGE.ALT })).toBeInTheDocument();
    });

    test('Available roles list Snapshots.', () => {
        const componentWithData = render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        ).container;
        const componentWithoutData = render(
            <AvailableRolesList
                roles={[]}
                addedRoles={[]}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        ).container;

        expect(componentWithData).toMatchSnapshot();
        expect(componentWithoutData).toMatchSnapshot();
    });

    test('Snapshots with warning messages about added roles in past.', () => {
        const utils = render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testAddedRolesList[0].name },
        });

        expect(utils).toMatchSnapshot();
    });

    test('Snapshots with messages about create new role.', () => {
        const utils = render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );
        const testInputValue = 'test_value_1234';

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testInputValue },
        });

        expect(utils).toMatchSnapshot();
    });

    test('Snapshots with warning messages about adding available role.', () => {
        const utils = render(
            <AvailableRolesList
                roles={testRolesList}
                addedRoles={testAddedRolesList}
                addRole={mockedAddRole}
                createNewRole={mockedCreateRole}
            />,
        );

        fireEvent.change(screen.getByRole('combobox'), {
            target: { value: testRolesList[0].name },
        });

        expect(utils).toMatchSnapshot();
    });
});
