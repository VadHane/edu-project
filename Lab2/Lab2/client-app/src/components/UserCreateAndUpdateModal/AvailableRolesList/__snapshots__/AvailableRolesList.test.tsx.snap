// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test available role list Available roles list Snapshots. 1`] = `
<div>
  <div>
    <div>
      <datalist
        id="available-roles-list"
      >
        <option
          value="TestRole1"
        />
        <option
          value="TestRole2"
        />
      </datalist>
      <input
        id="available-roles-list-input"
        list="available-roles-list"
        maxlength="30"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Test available role list Available roles list Snapshots. 2`] = `
<div>
  <div>
    <div>
      <datalist
        id="available-roles-list"
      />
      <input
        id="available-roles-list-input"
        list="available-roles-list"
        maxlength="30"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Test available role list Snapshots with messages about create new role. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <datalist
            id="available-roles-list"
          >
            <option
              value="TestRole1"
            />
            <option
              value="TestRole2"
            />
          </datalist>
          <input
            id="available-roles-list-input"
            list="available-roles-list"
            maxlength="30"
            type="text"
            value="test_value_1234"
          />
        </div>
        <div
          class="aprove-add-role"
        >
          <span>
            Would you like to create new role?
          </span>
           
          <br />
          <img
            alt="Approve"
            src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
          />
          <img
            alt="Cancel"
            src="https://cdn-icons-png.flaticon.com/512/1828/1828939.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <datalist
          id="available-roles-list"
        >
          <option
            value="TestRole1"
          />
          <option
            value="TestRole2"
          />
        </datalist>
        <input
          id="available-roles-list-input"
          list="available-roles-list"
          maxlength="30"
          type="text"
          value="test_value_1234"
        />
      </div>
      <div
        class="aprove-add-role"
      >
        <span>
          Would you like to create new role?
        </span>
         
        <br />
        <img
          alt="Approve"
          src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
        />
        <img
          alt="Cancel"
          src="https://cdn-icons-png.flaticon.com/512/1828/1828939.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test available role list Snapshots with warning messages about added roles in past. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <datalist
            id="available-roles-list"
          >
            <option
              value="TestRole1"
            />
            <option
              value="TestRole2"
            />
          </datalist>
          <input
            id="available-roles-list-input"
            list="available-roles-list"
            maxlength="30"
            type="text"
            value="TestAddedRole3"
          />
        </div>
        <div>
          <span>
            Roles list includes this role.
          </span>
           
          <br />
          <img
            alt="Approve"
            src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <datalist
          id="available-roles-list"
        >
          <option
            value="TestRole1"
          />
          <option
            value="TestRole2"
          />
        </datalist>
        <input
          id="available-roles-list-input"
          list="available-roles-list"
          maxlength="30"
          type="text"
          value="TestAddedRole3"
        />
      </div>
      <div>
        <span>
          Roles list includes this role.
        </span>
         
        <br />
        <img
          alt="Approve"
          src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test available role list Snapshots with warning messages about adding available role. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <datalist
            id="available-roles-list"
          >
            <option
              value="TestRole1"
            />
            <option
              value="TestRole2"
            />
          </datalist>
          <input
            id="available-roles-list-input"
            list="available-roles-list"
            maxlength="30"
            type="text"
            value="TestRole1"
          />
        </div>
        <div
          class="aprove-add-role"
        >
          <span>
            Would you like to add this role?
          </span>
           
          <br />
          <img
            alt="Approve"
            src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
          />
          <img
            alt="Cancel"
            src="https://cdn-icons-png.flaticon.com/512/1828/1828939.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <datalist
          id="available-roles-list"
        >
          <option
            value="TestRole1"
          />
          <option
            value="TestRole2"
          />
        </datalist>
        <input
          id="available-roles-list-input"
          list="available-roles-list"
          maxlength="30"
          type="text"
          value="TestRole1"
        />
      </div>
      <div
        class="aprove-add-role"
      >
        <span>
          Would you like to add this role?
        </span>
         
        <br />
        <img
          alt="Approve"
          src="https://cdn-icons-png.flaticon.com/512/1828/1828743.png"
        />
        <img
          alt="Cancel"
          src="https://cdn-icons-png.flaticon.com/512/1828/1828939.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
